### DISCUSSION TITLE
How to dynamically free memory after a service call

### DISCUSSION THREAD

**Discussion by livingca** [ntex-rs/ntex#200] [Q&A] [answered]
I stress tested the interface and after the test, the memory of the service didn't come down, and I thought "Isn't this the right way to use this, built on ntex+rbatis+tokio."

**Comment by fakeshadow** [comment, marked as answer]
This is a common "issue" likely caused by your allocator. You can reference this [reddit post](https://www.reddit.com/r/rust/comments/ybu6gz/memory_leak_free_memory_not_being_reclaimed_what/) for explanation and possible fixes. 

In short memory not always return to OS immediately when your free it. They may kept around for better memory usage.
